gnocchi/carbonara.py

```diff
@@ -453,11 +453,8 @@ def __iter__(self):
         return self
 
     def __hash__(self):
-        return hash(
-            str(datetime64_to_epoch(self.key))
-            +
-            str(self.sampling / ONE_SECOND)
-        )
+        return hash(str(self.key.astype('datetime64[ns]')) +
```

Comment:
So it seems that it's better to use `view` than `astype`: `view` does not return a copy, where `astype` does (unless you pass `copy=False`, but then, just use `view`).